### PR TITLE
Bug fix for code gen and type equality for refinements

### PR DESCRIPTION
**Pull request by Cypher1**
Fixes schema 2 graph modeling of type equality. 

`[{field: Type}[f(field)]]` being considered eq to
`[{field: Type}[g(field)]]` where `f` and `g` can't be rewritten to one another.

### CODE REVIEW

**Comment by Cypher1** [comment]
I'm wondering if there's some work that we should do (now or later) to ensure that a refined handle and unrefined handle should share a base class in KT, but I'm not sure about the semantics of companion objects in KT (i.e. can a subclass have an overridden version of SCHEMA without breaking the parent class?).

Adding @mykmartin for review of schema2graph
@shans for review of schema changes
@jasonwyatt for review of kotlin code

Thanks

File: src/runtime/refiner.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export enum Primitive {
   UNKNOWN = '~query_arg_type',
 }
 
-export enum AtleastAsSpecific {
+export enum AtLeastAsSpecific {
```
Comment (Cypher1) [inline src/runtime/refiner.ts]: I'm also fixing this typo as I keep writing it 'wrong'.

**Comment by shans** [comment]
@csilvestrini could you look at Kotlin changes? (@Cypher1 jwf is away for a bit)

File: src/runtime/schema.ts [hunk 2]
```diff
@@ -156,34 +156,49 @@ export class Schema {
...
+    const res = Refinement.isAtLeastAsSpecificAs(this.refinement, otherSchema.refinement);
+    if (res === AtLeastAsSpecific.NO) {
+      return AtLeastAsSpecific.NO;
+    } else if (res === AtLeastAsSpecific.UNKNOWN) {
+      best = AtLeastAsSpecific.UNKNOWN;
+    }
+    return best;
+  }
+
+  isAtleastAsSpecificAs(otherSchema: Schema): boolean {
```
Comment (csilvestrini) [inline src/runtime/schema.ts]: Did you want to correct this one too? `isAtLeast...`

Comment (Cypher1) [reply src/runtime/schema.ts]: Thought my language server rename would have got them all. Resorted to `git grep` and found a bunch, thanks.

**Comment by Cypher1** [comment]
Copybara submit to 302819622

**Comment by Cypher1** [comment]
/gcbrun
^ Re-running due to timeout in wasm tests.

302819622 has completed successfully

**Pull request merged**